### PR TITLE
[chart] Document ksm defaults in the values

### PR DESCRIPTION
**Pull request by kang-makes**
Solves #416

### CODE REVIEW

File: charts/newrelic-infrastructure/values.yaml [hunk 1]
```diff
@@ -140,16 +140,16 @@ ksm:
...
-  # -- the port is not discovered and the specified one is used
-  # port: 8080
-  # -- only the pods matching this selector are taken into account during autodiscovery
-  # selector: "app.kubernetes.io/name=kube-state-metrics"
-  # -- only the pods matching this namespace taken into account during autodiscovery
-  # namespace: "ksm-namespace"
+    # -- if specified autodiscovery is not performed and the specified URL is used
+    # staticUrl: "http://test.io:8080/metrics"
+    # -- pods matching this selector are taken into account during autodiscovery
+    selector: "app.kubernetes.io/name=kube-state-metrics"
```
Comment (roobre) [inline charts/newrelic-infrastructure/values.yaml:146]:
```suggestion
    # -- Label selector that will be used to automatically discover an instance of kube-state-metrics running in the cluster.
    selector: "app.kubernetes.io/name=kube-state-metrics"
```

File: charts/newrelic-infrastructure/values.yaml [hunk 2]
```diff
@@ -140,16 +140,16 @@ ksm:
...
+    # -- the URL scheme cannot be discovered and so you have to specify one
+    scheme: "http"
```
Comment (roobre) [inline charts/newrelic-infrastructure/values.yaml:148]:
```suggestion
    # -- Scheme to use to connect to kube-state-metrics. Supported values are `http` and `https`.
    scheme: "http"
```

File: charts/newrelic-infrastructure/values.yaml [hunk 3]
```diff
@@ -140,16 +140,16 @@ ksm:
...
+    # -- the port is not discovered and the specified one is used
+    # port: 8080
+    # -- only the pods matching this namespace taken into account during autodiscovery
+    # namespace: "ksm-namespace"
```
Comment (roobre) [inline charts/newrelic-infrastructure/values.yaml:152]:
```suggestion
    # -- Restrict autodiscovery of the kube-state-metrics service to a particular namespace.
    # @default -- All namespaces are searched (recommended).
    # namespace: "ksm-namespace"
```

**Review by roobre — COMMENTED** [review]
Some minor things on wording

File: charts/newrelic-infrastructure/values.yaml [hunk 4]
```diff
@@ -140,16 +140,17 @@ ksm:
...
+    # staticUrl: "http://test.io:8080/metrics"
+    # -- Label selector that will be used to automatically discover an instance of kube-state-metrics running in the cluster.
+    selector: "app.kubernetes.io/name=kube-state-metrics"
+    # -- Scheme to use to connect to kube-state-metrics. Supported values are `http` and `https`.
+    scheme: "http"
+    # -- Restrict autodiscovery of the kube-state-metrics endpoint to those using a specific port. If empty or `0`, all endpoints are considered regardless of their port (recommended).
+    # port: 8080
+    # -- Restrict autodiscovery of the kube-state-metrics service to a particular namespace.
+    # @default -- All namespaces are searched (recommended).
+    # namespace: "ksm-namespace"
```
Comment (paologallinaharbur) [inline charts/newrelic-infrastructure/values.yaml:153]: if uncommented and empty wouldn't appear in the values? Same for staticUrl

Comment (kang-makes) [reply charts/newrelic-infrastructure/values.yaml:153]: Will the integration behave properly? I would not like to send `nil` to the integration.

File: charts/newrelic-infrastructure/templates/ksm/scraper-configmap.yaml [hunk 5]
```diff
@@ -11,6 +11,5 @@ data:
   nri-kubernetes.yml: |-
     {{- (merge .Values.common.config (include "newrelic.integrationConfigDefaults" . | fromYaml)) | toYaml | nindent 4 }}
     ksm:
-      {{- /* Notice that new values takes precedence since they have no default and the user need to set them up specifically*/ -}}
-      {{- mustMergeOverwrite (include "newrelic.compatibility.ksm.legacyData" . | fromYaml) .Values.ksm.config | toYaml | nindent 6 -}}
+      {{- mustMergeOverwrite .Values.ksm.config  (include "newrelic.compatibility.ksm.legacyData" . | fromYaml) | toYaml | nindent 6 -}}
```
[the remark this answers is not in the document]
Comment (kang-makes) [reply charts/newrelic-infrastructure/templates/ksm/scraper-configmap.yaml:14]: Legacy should have precedence as we are in compatibility mode. The user should do the changes to the new one to standardize its configuration.

File: charts/newrelic-infrastructure/values.yaml (hunk 4 above)
Comment (kang-makes) [reply charts/newrelic-infrastructure/values.yaml:153]: Also, that is for advanced users that go to the values to look for advanced use cases.

**Pull request merged**